### PR TITLE
Fix reCAPTCHA API URL

### LINKED ISSUE #7870: Recaptcha doesn't work
- Gitea version (or commit ref): 1.9.1
- Git version: 2.20.1
- Operating system: Debian GNU/Linux 10 (Buster)
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://gitfesta.com:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist: gitea.log: https://gist.github.com/mzch/464588c5e49c9841876d10222d04bbc3
                  daemon.log: https://gist.github.com/mzch/e0fb7674cf26d38ce4073d19b4670266

## Description
Using ReCaptcha, gitea shows captcha_incorrect
...

## Screenshots
FireShot_Capture_003

### PR DESCRIPTION
**Pull request by mzch**
Fixes  #7870

### CODE REVIEW

**Review by techknowlogick — DISMISSED** [review]
The default value is `RECAPTCHA_URL = https://www.google.com/recaptcha/` so this will fail for those who have set this to something else.

**Comment by baxandrei** [comment]
@techknowlogick see this: https://github.com/go-gitea/gitea/issues/7870#issuecomment-555566306

File: modules/recaptcha/recaptcha.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ type Response struct {
 	ErrorCodes  []string  `json:"error-codes"`
 }
 
-const apiURL = "/api/siteverify"
+const apiURL = "/recaptcha/api/siteverify"
```
Comment (lafriks) [inline modules/recaptcha/recaptcha.go]:
```suggestion
const apiURL = "api/siteverify"
```

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by lafriks** [comment]
You are right, there is bug, see:
https://play.golang.org/p/9wG1S8TvS33

I recommend in review correct solution

**Review by lafriks — APPROVED** [review]

**Review by jolheiser — APPROVED** [review]

**Comment by zeripath** [comment]
Oh damn. Mea culpa. Could we fix urljoin instead/aswell? It's crazy that it's doing this. It absolutely should not be doing this as it will likely be causing other bugs too.

**Comment by mzch** [comment]
I found another fix.

**Comment by lafriks** [comment]
@zeripath it is doing everything correctly imho

**Comment by baxandrei** [comment]
I'm curious to see the solution, where can I find it?

**Comment by mzch** [comment]
@baxandrei I think it comes from specs of ResolveReference()

**Pull request closed without merging**